hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -605,4 +609,61 @@ void removeMappings() {
     public Operation getPostJoinOperation() {
         return ownershipMappings.isEmpty() ? null : new PostJoinClientOperation(ownershipMappings);
     }
+
+    @Override
+    public Map<ClientType, Integer> getConnectedClientStats() {
+
+        int numberOfCppClients    = 0;
+        int numberOfDotNetClients = 0;
+        int numberOfJavaClients   = 0;
+        int numberOfOtherClients  = 0;
+
+        Operation clientInfoOperation = new GetConnectedClientsOperation();
+        OperationService operationService = node.nodeEngine.getOperationService();
+        Map<ClientType, Integer> resultMap = new HashMap<ClientType, Integer>();
+        Map<String, ClientType> clientsMap = new HashMap<String, ClientType>();
+
+        for (MemberImpl member : node.getClusterService().getMemberList()) {
+            Address target = member.getAddress();
+            Future<Map<String, ClientType>> future
+                    = operationService.invokeOnTarget(SERVICE_NAME, clientInfoOperation, target);
+            try {
+                Map<String, ClientType> endpoints = future.get();
+                if (endpoints == null) {
+                    continue;
+                }
+                //Merge connected clients according to their uuid.
+                for (Map.Entry<String, ClientType> entry : endpoints.entrySet()) {
+                    clientsMap.put(entry.getKey(), entry.getValue());
+                }
+            } catch (Exception e) {
+                logger.warning("Cannot get client information from: " + target.toString(), e);
+            }
+        }
+
+        //Now we are regrouping according to the client type
+        for (Map.Entry<String, ClientType> entry : clientsMap.entrySet()) {
```

Comment:
You can iterate just `values()` here instead of entries. So, you'll avoid created garbage objects.

```java
for (ClientType clientType : clientsMap.values()) {
...
```